src/lib/api/client.ts

```diff
@@ -0,0 +1,49 @@
+import axios, { AxiosError, AxiosRequestConfig, AxiosResponse } from 'axios'
+
+const instance = axios.create({
+  baseURL: process.env.NEXT_PUBLIC_API_URL,
+})
+
+const onRequestFulfilled = (config: AxiosRequestConfig) => {
+  return {
+    ...config,
+    headers: {
+      Authorization: 'Bearer TOKEN',
+    },
+  }
+}
+
+const onResponseFulfilled = (res: AxiosResponse) => {
+  if (res.data.result === 'SUCCESS') {
+    return res.data.data
+  }
+
+  return Promise.reject(res.data)
+}
+
+const onResponseRejected = (error: AxiosError) => {
+  if (error.response?.data?.message != null) {
+    return Promise.reject(new Error(error.response.data.message))
+  }
+
+  return Promise.reject(error)
```

Comment:
마이너) 이건 취향 차이이기는 하지만 요렇게 갈 수도 있을 것 같기는 하네용

```suggestion
  return Promise.reject(new Error(error.response?.data?.message ?? error))
```

Reply:
`??` 삼항연산자가 획기적으로 짧아지는 마법.. 정말 좋아합니다ㅎㅎ